### PR TITLE
Added function getAllCurrencies

### PR DESCRIPTION
**Pull request by MathewWith**
Added a getAllCurrencies function

### CODE REVIEW

File: src/helpers/getAllCurrencies.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import axios from 'axios'
+
+export const getAllCurrencies = async (setAllCurrencies: any) => {
```
Comment (CodingRuslan) [inline src/helpers/getAllCurrencies.ts]: getAllCurrencies it's not helper function. This func should be store in service file

File: src/helpers/getAllCurrencies.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+    return await axios.get('https://openexchangerates.org/api/currencies.json')
```
Comment (CodingRuslan) [inline src/helpers/getAllCurrencies.ts]: You should write pure functions. getAllCurrencies should works only with getting data from api.
Read about Pure Functions

File: src/App.tsx [hunk 3]
```diff
@@ -1,10 +1,17 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import './App.scss';
+import { getAllCurrencies } from './helpers/getAllCurrencies';
 
 function App() {
+  const [allCurrencies, setAllCurrencies] = useState([])
+
+  useEffect(() => {
+    getAllCurrencies(setAllCurrencies)
```
Comment (CodingRuslan) [inline src/App.tsx]:
```suggestion
   const result = await getAllCurrencies()
   setAllCurrencies(result)
```

File: src/helpers/getAllCurrencies.ts (hunk 1 above)
Comment (MathewWith) [reply src/helpers/getAllCurrencies.ts]: Okey

File: src/App.tsx [hunk 4]
```diff
@@ -1,17 +1,21 @@
...
+import { getAllCurrencies } from './service/getAllCurrencies';
 
 function App() {
-  const [allCurrencies, setAllCurrencies] = useState([])
+  const [allCurrencies, setAllCurrencies] = useState([''])
 
   useEffect(() => {
-    getAllCurrencies(setAllCurrencies)
+    const res = async () => {
+      const data = await getAllCurrencies()
```
Comment (CodingRuslan) [inline src/App.tsx]: What is data ?  Name your variable more clear

File: src/App.tsx [hunk 5]
```diff
@@ -1,17 +1,21 @@
...
-import { getAllCurrencies } from './helpers/getAllCurrencies';
+import { getAllCurrencies } from './service/getAllCurrencies';
 
 function App() {
-  const [allCurrencies, setAllCurrencies] = useState([])
+  const [allCurrencies, setAllCurrencies] = useState([''])
 
   useEffect(() => {
-    getAllCurrencies(setAllCurrencies)
+    const res = async () => {
```
Comment (CodingRuslan) [inline src/App.tsx]: What is res ?

File: src/service/getAllCurrencies.ts [hunk 6]
```diff
@@ -1,9 +1,9 @@
 import axios from 'axios'
 
-export const getAllCurrencies = async (setAllCurrencies: any) => {
+export const getAllCurrencies = async () => {
     return await axios.get('https://openexchangerates.org/api/currencies.json')
-    .then(data => {
+    .then(function (data){
```
Comment (CodingRuslan) [inline src/service/getAllCurrencies.ts]: Why don't use arrow function ?

File: src/App.tsx [hunk 7]
```diff
@@ -1,10 +1,21 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import './App.scss';
+import { getAllCurrencies } from './service/getAllCurrencies';
 
 function App() {
+  const [allCurrencies, setAllCurrencies] = useState([''])
```
Comment (CodingRuslan) [inline src/App.tsx]: What is type of this useState ?

File: src/service/getAllCurrencies.ts [hunk 8]
```diff
@@ -0,0 +1,9 @@
+import axios from 'axios'
+
+export const getAllCurrencies = async () => {
```
Comment (CodingRuslan) [inline src/service/getAllCurrencies.ts]: Use output type for functions

File: src/service/getAllCurrencies.ts [hunk 9]
```diff
@@ -0,0 +1,9 @@
...
+    return await axios.get('https://openexchangerates.org/api/currencies.json')
+    .then(function (data){
+        let arr = Object.keys(data.data)
```
Comment (CodingRuslan) [inline src/service/getAllCurrencies.ts]: arr ? Read Clean code and name variable more clear

File: src/service/getAllCurrencies.ts [hunk 10]
```diff
@@ -0,0 +1,9 @@
+import axios from 'axios'
```
Comment (CodingRuslan) [inline src/service/getAllCurrencies.ts]: Do you want to have one service for one endpoint ?

**Review by CodingRuslan — CHANGES_REQUESTED** [review]

File: src/App.tsx (hunk 7 above)
Comment (MathewWith) [reply src/App.tsx]: I show the program that allCurrencies is string array

Comment (MathewWith) [reply src/App.tsx:6]: Can I do it different?

File: src/service/getAllCurrencies.ts (hunk 10 above)
Comment (MathewWith) [reply src/service/getAllCurrencies.ts]: I think I can add folder getAllCurrencies, correct?

File: src/App.tsx (hunk 7 above)
Comment (CodingRuslan) [reply src/App.tsx]: May be apply typescript features and added the type of state ?

File: src/App.tsx [hunk 11]
```diff
@@ -1,10 +1,33 @@
...
 
 function App() {
+  
+  const [allCurrencies, setAllCurrencies] = useState([''])
+  const [leftInputValue, setLeftInputValue] = useState('')
+  const [rightInputValue, setRightInputValue] = useState('')
+  const [isReversed, setIsReversed] = useState(false)
+
+  useEffect(() => {
+    const getArrayCurrencies = async () => {
```
Comment (CodingRuslan) [inline src/App.tsx]: Don't name variavles like Array, List, Number and so on

File: src/App.tsx [hunk 12]
```diff
@@ -1,10 +1,33 @@
...
+      const arrayCurrencies = await getAllCurrencies()
```
Comment (CodingRuslan) [inline src/App.tsx]: [Read clean code! ](https://medium.com/@pabashani.herath/clean-code-naming-conventions-4cac223de3c6)

File: src/App.tsx [hunk 13]
```diff
@@ -1,10 +1,33 @@
...
+    }
+    getArrayCurrencies()
+  }, [])
+
   return (
     <div className="App">
-      App
+      <h1>Change money</h1>
+
+      {ListCurrencies(allCurrencies)}
```
Comment (CodingRuslan) [inline src/App.tsx]: Why you use function calling instead of components and props ?

File: src/components/List-currencies/List-currencies.tsx [hunk 14]
```diff
@@ -0,0 +1,15 @@
+import {useState} from "react"
+import { Select } from "../Select/Select"
+
+export const ListCurrencies: any = ({allCurrencies}: {allCurrencies: string[]}) => {
```
Comment (CodingRuslan) [inline src/components/List-currencies/List-currencies.tsx]: any ?

File: src/components/Select/Select.tsx [hunk 15]
```diff
@@ -0,0 +1,14 @@
+
+export const Select = (props: any) => {
```
Comment (CodingRuslan) [inline src/components/Select/Select.tsx]: any ?

File: src/components/Select/Select.tsx [hunk 16]
```diff
@@ -0,0 +1,14 @@
...
+    const {allCurrencies} = props;
+    const {setCurrency} = props;
```
Comment (CodingRuslan) [inline src/components/Select/Select.tsx]: Combine this props

File: src/components/Select/Select.tsx [hunk 17]
```diff
@@ -0,0 +1,14 @@
+
+export const Select = (props: any) => {
+    const {allCurrencies} = props;
+    const {setCurrency} = props; 
+    return (
+        <select onChange={(e) => setCurrency(e.target.value)}>
+                {
+                    allCurrencies.map((cur: string) => {
```
Comment (CodingRuslan) [inline src/components/Select/Select.tsx]: cur ?

File: src/service/getAllCurrencies.ts (hunk 10 above)
Comment (CodingRuslan) [reply src/service/getAllCurrencies.ts]: no

File: src/service/getAllCurrencies.ts [hunk 18]
```diff
@@ -0,0 +1,9 @@
...
+
+export const getAllCurrencies = async (): Promise<string[]> => {
+    return await axios.get('https://openexchangerates.org/api/currencies.json')
+    .then((data) => {
```
Comment (CodingRuslan) [inline src/service/getAllCurrencies.ts]: then ?

**Review by CodingRuslan — CHANGES_REQUESTED** [review]

File: src/App.tsx [hunk 19]
```diff
@@ -1,10 +1,33 @@
...
+    }
+    getCurrencies()
+  }, [])
+
   return (
     <div className="App">
-      App
+      <h1>Change money</h1>
+
+      {<ListCurrencies allCurrencies={allCurrencies} />}
```
Comment (CodingRuslan) [inline src/App.tsx]: Why do you use brackets ?

**Review by CodingRuslan — CHANGES_REQUESTED** [review]

File: src/App.tsx (hunk 19 above)
Comment (MathewWith) [reply src/App.tsx]: My silly mistake. Already fixed it

**Pull request merged**